### ISSUE TITLE
Feature Request: Supplement forward geocoding search results with another data source

### ISSUE THREAD

**Issue by weifageo** [mapbox/mapbox-search-android-examples#28]
Hello there Mapbox Team,

I've recently been playing around with the Search SDK on Android and was wondering if it would be possible to port over the localGeocoder logic used by your Mapbox GL Geocoder.  I'm building an app for my company and would like to be able to Search for some of our own data points by name.  On the Web environment, we were able to achieve this using a localGeocoder as explained in this example: https://docs.mapbox.com/mapbox-gl-js/example/forward-geocode-custom-data/

Just curious if something like this maybe possible for the Android Search SDK moving forward?  Thanks for your support!

**Comment by friendoye** [comment]
Hello @weifageo 👋 

As for now we allow users to specify extra data only via `FavoritesDataProvider` (`MapboxSearchSdk.serviceProvider.favoritesDataProvider()`) and `HistoryDataProvider` (`MapboxSearchSdk.serviceProvider.historyDataProvider()`). Please note that changing content of these providers will affect some parts of our UI from `mapbox-search-android-ui` artifact (for example, content of "Favorite" tab).

We have plans to provide support for custom data providers in the nearest future. Will keep you in touch 👌

**Comment by friendoye** [comment]
Hey @weifageo!

Sorry for the long response. All I can say work on this feature request is currently in progress, no ETA for now. Will leave a comment here once new version of Search SDK with this feature is published.

**Comment by weifageo** [comment]
That sounds great @friendoye, thanks for your support!

**Comment by DzmitryFomchyn** [comment]
@weifageo functionality has been added in the Search SDK [v1.0.0-beta.19](https://github.com/mapbox/mapbox-search-android-examples/releases/tag/v1.0.0-beta.19).

To get more information see code examples: for [Kotlin](https://github.com/mapbox/mapbox-search-android-examples/blob/main/app/src/main/java/com/mapbox/search/sample/api/CustomIndexableDataProviderKotlinExample.kt) and for [Java](https://github.com/mapbox/mapbox-search-android-examples/blob/main/app/src/main/java/com/mapbox/search/sample/api/CustomIndexableDataProviderJavaExample.java).

Let me know if you have any questions, we're looking forward to your feedback!

**Comment by weifageo** [comment]
Thank you for the heads up, I will give the code examples a try and will report back shortly!

**Comment by weifageo** [comment]
Thanks for getting back to me!  After updating to the latest v1.0.0-beta.21 from v1.0.0-beta.19 the getMainExecutor() error has been resolved!

Sorry for the confusion on my part, but I'm not sure where to implement this MapboxSearchSdk statement you've provided above with my current SearchBottomSheetView implementation.  Here is how I've setup the current SearchBottomSheetView within the onCreate function of my MainActivity.

`// Search
        searchBottomSheetView = findViewById(R.id.search_view);
        myHotCategories.add(Category.PARKS);
        myHotCategories.add(Category.GAS_STATION);
        myHotCategories.add(Category.PARKING);
        myHotCategories.add(Category.HOTEL);
        SearchBottomSheetView.Configuration myConfiguration = new SearchBottomSheetView.Configuration.Builder().hotCategories(myHotCategories).build();
        searchBottomSheetView.initializeSearch(savedInstanceState, myConfiguration);
        searchBottomSheetView.hide();
        searchBottomSheetView.setHideableByDrag(true);`

Any additional advice is much appreciated, thanks!

**Comment by friendoye** [comment]
> Sorry for the confusion on my part, but I'm not sure where to implement this MapboxSearchSdk statement you've provided above with my current SearchBottomSheetView implementation.

The main advise is to invoke

```java
MapboxSearchSdk.getServiceProvider().globalDataProvidersRegistry().register(...)
```

function as soon as possible before you show `SearchBottomSheetView` on the screen. I suggest to put this line right after Search SDK initialization logic:

```java
MapboxSearchSdk.initialize(...)
MapboxSearchSdk.getServiceProvider().globalDataProvidersRegistry().register(...)
```

**Comment by weifageo** [comment]
Thanks for getting back to me! I was able to get the Custom POIs to show within the SearchBottomSheetView!!!  Thanks very much for your support!

Now that I've got this basic implementation working with a few predefined POI items, I'm just wondering what you would recommend if I needed to add a few hundred or thousand POIs?  Do they all need to be manually prebuilt into the InMemoryDataProvider with createRecord?

Any more advice you have is much appreciated, thanks again!
